src/server/pfs/server/val_server.go

```diff
@@ -217,34 +191,68 @@ func (a *validatedAPIServer) DiffFile(request *pfs.DiffFileRequest, server pfs.A
 	if request.NewFile == nil {
 		return errors.New("file cannot be nil")
 	}
-	if request.NewFile.Commit == nil {
-		return errors.New("file commit cannot be nil")
-	}
-	request.NewFile.Commit.Repo = request.NewFile.Commit.AccessRepo()
-	if request.NewFile.Commit.Repo == nil {
-		return errors.Errorf("new file's commit must have a repo")
+	if err := checkCommit(request.NewFile.Commit); err != nil {
+		return errors.Wrap(err, "check new file commit")
 	}
 	if request.OldFile != nil {
-		request.OldFile.Commit.Repo = request.OldFile.Commit.AccessRepo()
-		if request.OldFile.Commit.Repo == nil {
-			return errors.Errorf("old file's commit must have a repo")
+		if err := checkCommit(request.OldFile.Commit); err != nil {
+			return errors.Wrap(err, "check old file commit")
 		}
 	}
 	return a.apiServer.DiffFile(request, server)
 }
 
+func (a *validatedAPIServer) AddFileSet(ctx context.Context, req *pfs.AddFileSetRequest) (_ *types.Empty, retErr error) {
+	if err := checkCommit(req.Commit); err != nil {
+		return nil, err
+	}
+	return a.apiServer.AddFileSet(ctx, req)
+}
+
+func (a *validatedAPIServer) SubscribeCommit(request *pfs.SubscribeCommitRequest, stream pfs.API_SubscribeCommitServer) (retErr error) {
+	if request.From != nil {
```

Comment:
Do we need this?  There is already a nil check in `checkCommit`

Reply:
request.From is not required, so we don't want to error out on a nil commit here